src/Uno.UI/UI/Xaml/Internal/InputManager.cs

```diff
@@ -66,4 +87,80 @@ internal bool LastInputWasNonFocusNavigationKeyFromSIP()
 		//TODO Uno: Implement
 		return false;
 	}
+
+	internal void RedirectPointer(Windows.UI.Input.PointerPoint pointer, InteractionTracker tracker)
+	{
+		if (_pointerRedirections?.TryGetValue(pointer.PointerId, out var recognizer) == true)
+		{
+			((GestureRecognizerOwner)recognizer.Owner).Trackers.Add(tracker);
+			return;
+		}
+
+		_pointerRedirections ??= new();
+		recognizer = new GestureRecognizer(new GestureRecognizerOwner([tracker], this));
+		recognizer.GestureSettings = GestureSettingsHelper.Manipulations;
+		recognizer.ManipulationStarting += OnRecognizerManipulationStarting;
+		recognizer.ManipulationStarted += OnRecognizerManipulationStarted;
+		recognizer.ManipulationUpdated += OnRecognizerManipulationUpdated;
+		recognizer.ManipulationInertiaStarting += OnRecognizerManipulationInertiaStarting;
+		recognizer.ManipulationCompleted += OnRecognizerManipulationCompleted;
+
+#if HAS_UNO_WINUI
+		recognizer.ProcessDownEvent(new PointerPoint(pointer));
+#else
+		recognizer.ProcessDownEvent(new PointerPoint(
+			pointer.FrameId, pointer.Timestamp, pointer.PointerDevice, pointer.PointerId, pointer.RawPosition, pointer.Position, pointer.IsInContact, new PointerPointProperties(pointer.Properties)));
+#endif
+		// TODO: Currently this dictionary is only used for UNO_HAS_MANAGED_POINTERS.
+		_pointerRedirections[pointer.PointerId] = recognizer;
+	}
+
+	private static readonly TypedEventHandler<GestureRecognizer, ManipulationStartingEventArgs> OnRecognizerManipulationStarting = (sender, args) =>
+	{
+		// TODO: Make sure ManipulationStarting is fired on UIElement.
+		var trackers = ((GestureRecognizerOwner)sender.Owner).Trackers;
+		foreach (var tracker in trackers)
+		{
+			tracker.StartUserManipulation();
+		}
+	};
+
+	private static readonly TypedEventHandler<GestureRecognizer, ManipulationStartedEventArgs> OnRecognizerManipulationStarted = (sender, args) =>
+	{
+#if UNO_HAS_MANAGED_POINTERS
```

Comment:
I think not even init anything if not supported, we should just ignore the redirection.

Reply:
It happened naturally when refactoring everything into PointerManager since PointerManager is only defined for `UNO_HAS_MANAGED_POINTERS` :)